amd_openvx/openvx/hipvx/vision_kernels.cpp

```diff
@@ -36,13 +36,19 @@ Hip_CannySobel_U16_U8_3x3_L1NORM(uint dstWidth, uint dstHeight,
     int y = hipBlockDim_y * hipBlockIdx_y + hipThreadIdx_y;
     bool valid = (x < dstWidth) && (y < dstHeight);
 
+    if (x >= dstWidth || y >= dstHeight) return;
```

Comment:
doing this will not compute values if x which exceeds after multiple of 8 values. For eg. ImageWidth not multiple of 8, the last reminder of values will not get computed